spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/dsl/XDDSLMessages.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.core.dsl;
+
+import static org.springframework.cloud.data.core.dsl.XDDSLMessages.Kind.ERROR;
+
+import java.text.MessageFormat;
+
+/**
+ * Contains all the messages that can be produced during Spring XD DSL
+ * parsing. Each message has a kind (info, warn, error) and a code number.
+ * Tests can be written to expect particular code numbers rather than
+ * particular text, enabling the message text to more easily be modified
+ * and the tests to run successfully in different locales.
+ * <p>
+ * When a message is formatted, it will have this kind of form
+ *
+ * <pre class="code">
+ * XD105E: (pos 34): Expected an argument value but was ' '
+ * </pre>
+ *
+ * </code> The prefix captures the code and the error kind, whilst the position is included if it is known.
+ *
+ * @author Andy Clement
+ */
+public enum XDDSLMessages {
```

Comment:
We need to remove "XD" from this class, starting with the name. Also, the name should not be plural due to the fact that an instance of this class is a single message. Maybe a good name would be `DSLErrorMessage`.